### PR TITLE
fix: redact headers, cookies, url params in har file

### PR DESCRIPTION
**Pull request by dkwon17**
### What does this PR do?
Extension to the https://github.com/eclipse/che/pull/22526 feature, that redacts auth headers, cookies, and params from har files.

### How to test this PR?

1. On an OpenShift cluster, install Eclipse Che
2. Checkout this PR, cd to `tests/e2e`
3. Run the following to set the necessary environment variables:
```
export TS_SELENIUM_BASE_URL=<CHE_URL>
export TS_SELENIUM_OCP_USERNAME=<USERNAME>
export TS_SELENIUM_OCP_PASSWORD=<PASSWORD>
export TS_SELENIUM_VALUE_OPENSHIFT_OAUTH=true
``` 
4. Run `npm ci`
5. Apply the following patch to guarantee that one of the smoke tests fail:
```
git apply <<EOF
diff --git a/tests/e2e/specs/SmokeTest.spec.ts b/tests/e2e/specs/SmokeTest.spec.ts
index 4a5f08bf76..709371a396 100644
--- a/tests/e2e/specs/SmokeTest.spec.ts
+++ b/tests/e2e/specs/SmokeTest.spec.ts
@@ -46,6 +46,7 @@ suite('The SmokeTest userstory', function (): void {
 		test('Check a project folder has been created', async function (): Promise<void> {
 			const projectName: string = FACTORY_TEST_CONSTANTS.TS_SELENIUM_PROJECT_NAME || StringUtil.getProjectNameFromGitUrl(factoryUrl);
 			projectSection = (await new SideBarView().getContent().getSections())[0]; // get the (WORKSPACE) section from the sidebar - contains project content
+			expect(false).to.be.true;
 			expect(await projectSection.findItem(projectName)).not.eqls(undefined);
 		});
 		test('Check the project files was imported', async function (): Promise<void> {
EOF
```
6. Run the smoke tests
```
export USERSTORY=SmokeTest && npm run test
```
7. The test should fail and a `.har` file should be available in `tests/e2e/report/<test-name>/`
8. Verify that the har file does not contain authentication headers, cookies, url params

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: tests/e2e/utils/StringUtil.ts [hunk 1]
```diff
@@ -59,4 +59,32 @@ export class StringUtil {
...
 	}
+
+	/**
+	 * replaces the cookie value of the specified cookie
+	 * @param cookie cookie names and values, seperated with ;
+	 * @param name name of cookie to replace its value for
+	 * @param replaceStr the new value of the cookie
+	 * @return updated cookie string with the cookie value replaced
+	 */
+	static updateCookieValue(cookie: string, name: string, replaceStr: string): string {
```
Comment (dkwon17) [inline tests/e2e/utils/StringUtil.ts:70]: @nallikaea I have made changes to use `StringUtil`

Comment (nallikaea) [reply tests/e2e/utils/StringUtil.ts:70]: Thank you!

File: tests/e2e/utils/CheReporter.ts [hunk 2]
```diff
@@ -181,12 +181,48 @@ class CheReporter extends mocha.reporters.Spec {
 			const networkLogsEntries: logging.Entry[] = await this.driverHelper.getDriver().manage().logs().get('performance');
 			const events: any[] = networkLogsEntries.map((entry): any[] => JSON.parse(entry.message).message);
 			const har: any = chromeHar.harFromMessages(events, { includeTextFromResponseBody: true });
+			try {
+				this.maskHarContent(har);
+			} catch (e) {
+				console.log(e);
```
Comment (nallikaea) [inline tests/e2e/utils/CheReporter.ts]: can be `Logger.err(e)`

**Review by nallikaea — APPROVED** [review]
Approved with small comment

File: tests/e2e/utils/CheReporter.ts (hunk 2 above)
Comment (dkwon17) [reply tests/e2e/utils/CheReporter.ts]: Oops, the try catch was something I forgot to remove, can we just remove it @nallikaea ?

Comment (dkwon17) [reply tests/e2e/utils/CheReporter.ts]: I removed it, but please let me know if you would like to keep it

**Pull request merged**